src/modules/people-list/people-list-item.tsx

```diff
@@ -0,0 +1,203 @@
+
+const StyleOtherDetailWrapper = Styled.View`
+  width:100%
+`;
+
+const StyleHeadingUserNameWrapper = Styled(Heading)`
+  font-size: ${minorScale(3.6)}px;
+  width: 80%;
+`;
+
+const StyleUserIdTextWrapper = Styled(Text)`
+  font-size: ${minorScale(2.8)}px;
+  color: ${({theme}) => {
+    return theme.colors.grayFour;
+  }};
+  text-align:right
+`;
+
+const StyleUserDescriptionWrapper = Styled(Text)`
+  font-size: ${minorScale(2.8)}px;
+  color: ${({theme}) => {
+    return theme.colors.grayFour;
+  }};
+  margin-top: ${minorScale(1)}px;
+  margin-bottom: ${minorScale(1)}px;
+`;
+
+const StyleUserProfilePicWrapper = Styled(Avatar)`
+  testID:profile-pic`;
+
+const StyleLinkWrapper = Styled.TouchableOpacity``;
+
+export const PeopleListItem: React.FC<IProps> = ({
+  onUserClicked,
+  name,
+  maritalStatus,
+  gender,
+  churchEntityName,
+  profilePic,
+  userId,
+  footerElement,
+  isShowCheckbox,
+  onCheckboxPress,
+  isChecked,
+  leftButtons,
+  rightButtons,
+}): React.ReactElement => {
+  /**
+   * get other data like maritalStatus,gender,churchEntName
+   * add data to array if data is not null and then create string using join
+   * @returns
+   */
+  const getOtherDetailsString = () => {
+    const arrOtherDet: string[] = [];
+    //churchEntityName
+    const churchEntName: string | undefined =
+      churchEntityName !== null ? churchEntityName : '';
+
+    //maritalStatus
+    const marStatus: string | undefined =
+      maritalStatus !== null ? maritalStatus : '';
+
+    //gender
+    const strGender: string | undefined =
+      gender === null
+        ? ''
+        : String(gender).toLowerCase() === 'f'
+        ? 'Female'
+        : 'Male';
+
+    if (strGender !== undefined && strGender.length > 0) {
+      arrOtherDet.push(strGender);
+    }
+
+    if (marStatus !== undefined && marStatus.length > 0) {
+      arrOtherDet.push(marStatus);
+    }
+
+    if (churchEntName !== undefined && churchEntName.length > 0) {
+      arrOtherDet.push(churchEntName);
+    }
+
+    return arrOtherDet;
+  };
+
+  const renderUserPic = () => {
+    return (
+      <StyleUserProfilePicWrapper
+        uri={profilePic!}
+        initials={getUserNameFirstLastCharacter()}
+        size={40}
+        radius="small"
+        // testID="profile-pic"
+      />
+    );
+  };
+
+  const getUserNameFirstLastCharacter = () => {
+    const arrName = name?.split(' ');
+    if (arrName?.length === 2) {
+      return `${arrName[0].charAt(0)}${arrName[1].charAt(0)}`;
+    }
+    return `${arrName![0].charAt(0)}${arrName![0].charAt(1)}`;
+  };
+
+  const renderCheckBoxOrPic = () => {
+    if (isShowCheckbox && isShowCheckbox === true) {
+      return (
+        <StyleProfilePicCheckboxWrapper>
+          <StyleCheckboxWrapper>
+            <Checkbox onPress={onCheckboxPress} isChecked={isChecked!} />
+          </StyleCheckboxWrapper>
+          {renderUserPic()}
+        </StyleProfilePicCheckboxWrapper>
+      );
+    }
+    return renderUserPic();
+  };
```

Comment:
`renderCheckBox-OR-Pic` means that this function would either render the checkbox or the photo but I don't think that is the case. Also, a separate render function for this seems like an overkill.

Reply:
I have created separate render function